ls1intum/Artemis
# `Development`: Decouple LocalCI and LocalVC

### Checklist
#### General


#### Server

### Motivation and Context


Currently, the LocalVC and LocalCI are highly coupled inside of the `LocalCIConnectorService`. This was causing issues for further development.

### Description

The code was moved to the `LocalVCServletService`. Instead of directly using the `LocalCITriggerService` we now you use the interface `ContinuousIntegrationTriggerService` for triggering the builds.

### Steps for Testing

Prerequisites:
- 1 Instructor

Create a few programming exercises and submit a few changes and check that the system still works as expected

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
